File: Tried/1478.ts

const input_1487 = `9
10
0`

//const fs = require('fs')
//const input = fs.readFileSync('/dev/stdin', 'utf-8')
const lines_1487 = input_1487.split('\n').map(Number)

function next_1487 () {
  return Number(lines_1487.shift())
}


function matrix_1487(n: number) {
  let list = Array.from({ length: n }, (_, i) => i + 1);
  const maxDigits = String(n).length; // 
  const spacing = 3;

  for (let i = 0; i < n; i++) {
    
    let output = list.map(num => {
      return String(num).padStart(maxDigits)
    }).join(' '.repeat(spacing - maxDigits + 1));
    
    console.log(output);
    list.pop();
    list.unshift(i + 2);
  }
}

let curr = next_1487()
while(curr != 0 ){
  matrix_1487(curr)
  curr = next_1487()
  if(curr != 0 ){
    console.log('')
  }
}

File: beginner/pagina05/1478/1478.py
    
while True:
    N = int(input())
        
    if N == 0:
        break
    
    def exibir_line(lista):
        print(" ".join(f"{n:3}" for n in lista))
        
    def espelhar_lista(lista):
        lista.pop()
        novo = lista[0] + 1
        lista.insert(0, novo)
        return lista
    
    curr = [int(x) for x in range(1, N + 1)]
    exibir_line(curr)
    i = 1
    while i < N:
        novo = espelhar_lista(curr)
        exibir_line(novo)
        i+=1
        curr = novo
        
    print()